src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -235,8 +237,7 @@ private IEnumerable<JsonPropertySerialization> CreatePropertySerializations()
                         valueSerialization,
                         property.IsRequired,
                         ShouldExcludeInWireSerialization(property, inputModelProperty),
-                        customSerializationMethodName: serializationMapping?.SerializationValueHook,
-                        customDeserializationMethodName: serializationMapping?.DeserializationValueHook,
+                        serializationMapping: serializationMapping,
```

Comment:
@ArcturusZhang I reworked this to pass the whole serializationMapping into PropertySerialization so that it can be used for bicep as well.

Reply:
well we kind of do not want to use this type here because `serializationMapping` here is considered as an "input model", and the serialization object we built here should be considered as "output model"

Reply:
I see your point, but the mapping contains metadata is used by the PropertySerialization. What if I encapsulated the part that contains the serialization hooks into a new record and instead pass that through?

Reply:
Potentially consider making the attribute into a dictionary/tuple input where users pass the key is the MRW format, and the value is the method name.